blackduck-artifactory-common/src/main/java/com/synopsys/integration/blackduck/artifactory/modules/cancel/ScanCancelDecider.java

```diff
@@ -35,24 +35,28 @@ public ScanCancelDecider(ScanModuleConfig scanModuleConfig, ScanPropertyService
     @Override
     public CancelDecision getCancelDecision(RepoPath repoPath) {
         boolean metadataBlockDisabled = FALSE.equals(scanModuleConfig.isMetadataBlockEnabled());
-        boolean shouldScanRepository = !scanModuleConfig.getRepos().contains(repoPath.getRepoKey());
-        ItemInfo itemInfo = artifactoryPAPIService.getItemInfo(repoPath);
-        if (metadataBlockDisabled || shouldScanRepository || itemInfo.isFolder()) {
+        boolean shouldNotScanRepository = !scanModuleConfig.getRepos().contains(repoPath.getRepoKey()); // TODO: This should be a call to another service
+        if (metadataBlockDisabled || shouldNotScanRepository) {
             return CancelDecision.NO_CANCELLATION();
         }
 
-        Optional<Result> scanResult = scanPropertyService.getScanResult(repoPath);
-        if (scanResult.isPresent() && Result.FAILURE.equals(scanResult.get())) {
-            return CancelDecision.CANCEL_DOWNLOAD(String.format("The artifact was not successfully scanned. Found result %s.", Result.FAILURE));
-        } else if (scanResult.isPresent() && Result.SUCCESS.equals(scanResult.get())) {
+        // Getting ItemInfo from a virtual repository causes an exception resulting in failed downloads. We must verify the repository before doing this check. (IARTH-434)
```

Comment:
Ticket name here is good, I personally would also attach my GitHub username and a date